core/controllers/editor.py

```diff
@@ -214,11 +208,9 @@ def put(self, exploration_id):
         except utils.ValidationError as e:
             raise self.InvalidInputException(e)
 
-        try:
```

Comment:
`get_user_exploration_data()`'s failure cases are handled by its decorator, i.e can_edit_exploration

Reply:
I don't see a reference to the ExplorationUserData model there. What if it's just not present for some reason? Should still catch this.

Reply:
But how can I create an exception here? It won't throw an exception if ExplorationUserData model is None -- it will work fine. I think the possible reasons for raising an exception is covered by its decorator.

Reply:
There's a line in that function that does:

        exp_user_data = user_models.ExplorationUserDataModel.get(
            user_id, exploration_id)

What happens if that model instance doesn't exist?

Reply:
It returns None and the code below it works fine even if the model is None

Reply:
Ah I see, thanks for explaining. Yes, what you did sounds fine, then.